### PR TITLE
refac: Segregate submodule `package.step.aggregations.transformations`

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Working on simplifying aggregations. This PR moves files from the aggregation` submodule into a new submodule `transformations`.

image

## References

* #415

### CODE REVIEW

File: source/databricks/calculation-engine/package/steps/aggregation/aggregators.py [hunk 1]
```diff
@@ -363,40 +359,41 @@ def _aggregate_per_ga(
...
 
-    return create_dataframe_from_aggregation_result_schema(result)
+    return T.create_dataframe_from_aggregation_result_schema(result)
 
 
 def _aggregate_sum_and_set_quality(
     result: DataFrame, quantity_col_name: str, group_by: list[str]
 ) -> DataFrame:
     result = result.na.fill(value=0, subset=[quantity_col_name])
+    qualities_col_name = "qualities"
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/steps/aggregation/aggregators.py:369]: Colname.qualities?

Comment (BjarkeMeier) [reply source/databricks/calculation-engine/package/steps/aggregation/aggregators.py:369]: It's only a local variable unused outside the function.

File: source/databricks/calculation-engine/package/steps/aggregation/aggregators.py [hunk 2]
```diff
@@ -149,35 +147,33 @@ def aggregate_net_exchange_per_ga(df: DataFrame) -> DataFrame:
...
 
 
 def aggregate_production_ga_brp_es(enriched_time_series: DataFrame) -> DataFrame:
-    return _aggregate_per_ga_and_brp_and_es(
-        enriched_time_series, MeteringPointType.production, None
-    )
+    return _aggregate(enriched_time_series, MeteringPointType.production, None)
 
 
 # Function to aggregate sum per grid area and energy supplier (step 12, 13 and 14)
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/steps/aggregation/aggregators.py:175]: delete comment?

Comment (BjarkeMeier) [reply source/databricks/calculation-engine/package/steps/aggregation/aggregators.py:175]: In another PR. It's completely unrelated to these changes 😄

Comment (BjarkeMeier) [reply source/databricks/calculation-engine/package/steps/aggregation/aggregators.py]: Oh, no. You're right. Removed.

File: source/databricks/calculation-engine/package/steps/aggregation/aggregators.py [hunk 3]
```diff
@@ -149,35 +147,33 @@ def aggregate_net_exchange_per_ga(df: DataFrame) -> DataFrame:
...
-def _aggregate_per_ga_and_brp_and_es(
+def _aggregate(
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/steps/aggregation/aggregators.py:175]: is this name better?

**Review by JesperGomes — APPROVED** [review]

File: source/databricks/calculation-engine/package/steps/aggregation/aggregators.py (hunk 3 above)
Comment (BjarkeMeier) [reply source/databricks/calculation-engine/package/steps/aggregation/aggregators.py]: Reverted as per our talk.

**Pull request merged**